### PR TITLE
Access data through an interface

### PR DESCRIPTION
**Pull request by darrylmelander**
Create a data provider interface (#30)

### CODE REVIEW

**Comment by darrylmelander** [comment]
Please create a merge commit for this one (don't squash).  I want to keep the refactoring separate from the new ruc implementation.

File: prescient/data/providers/dat_data_provider.py
Comment (bknueven) [inline prescient/data/providers/dat_data_provider.py:56]: We be annotating this list with the times (as date-time strings, I think) for which the time series data is for, and this should be a requirement for this interface. That way, when creating the SCED we can do string or datetime comparison operations to know what data to get out of the associated problems. This will be especially necessary for #37.

File: prescient/data/providers/dat_data_provider.py
Comment (bknueven) [inline prescient/data/providers/dat_data_provider.py:242]: Eventually we should make this logic somewhat generic, e.g., see https://github.com/grid-parity-exchange/Egret/blob/8a396c9191a5c8905ca93c3f9cf9b9f223a10d73/egret/model_library/transmission/tx_utils.py#L205

File: prescient/engine/egret/egret_plugin.py
Comment (bknueven) [inline prescient/engine/egret/egret_plugin.py:1109]: Similar to the comment above -- since these are carrying forward the same information I think it would be good to have it annotated in a single place.

File: prescient/util/math_utils.py
Comment (bknueven) [inline prescient/util/math_utils.py:22]: I might suggest replacing the implementation of this function with `math.isclose(v1,v2,rel_tol=0, abs_tol=tol)`, which of course is equivalent.

**Review by bknueven — APPROVED** [review]
Overall the interface is very helpful, and there's some nice clean-up of the modeling engine and data manager.

I left a few comments, most of which should either be addressed or tracked as issues. @darrylmelander, I'll let you merge this when ready.

File: prescient/util/math_utils.py
Comment (darrylmelander) [reply prescient/util/math_utils.py:22]: I wasn't aware of that function.  I removed our `within_tolerance` and replaced it with `math.isclose`.

File: prescient/data/providers/dat_data_provider.py
Comment (darrylmelander) [reply prescient/data/providers/dat_data_provider.py:242]: Right now there are 5 places that have to know what is being predicted by a forecast.  I like the idea of reducing this down to a single spot.  Preferably I'd like to maintain the readability of `egret_plugin._copy_time_perid_data()`.  Also, of those 5 places I mentioned, 3 of them would require some non-trivial rework to get it to use a centralized definition of forecast data (creating a sced instance, storing forecasts, and storing forecast error).  For those reasons I'm deferring this to a new issue (issue #48)

File: prescient/engine/egret/egret_plugin.py
Comment (darrylmelander) [reply prescient/engine/egret/egret_plugin.py:1109]: See issue #48

File: prescient/data/providers/dat_data_provider.py
Comment (darrylmelander) [reply prescient/data/providers/dat_data_provider.py:56]: See issue #49

**Pull request merged**
